### PR TITLE
Adds a pull request workflow fundamental and CC-BY copyright license and link

### PR DESCRIPTION
**Pull request by remarcmij**
I am also working on basing JavaScript2 homework on PRs and plan to do so for JavaScript3 too.

### CODE REVIEW

File: fundamentals/homework_pr.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+## Commit and Push
+
+For each week's homework repeat the following steps.
+
+1. If there our any outstanding changes from the previous week, either commit or discard them.
+
+2. Start afresh by resetting your local repository back to the `master` branch. This ensures that the new homework is not mixed up with the homework of previous weeks. (That previous homework will still be available in their respective branches.) Use this command:
+
+    ```
+    git checkout master
```
Comment (jasongwartz) [inline fundamentals/homework_pr.md]: :clap:

File: fundamentals/homework_pr.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+3. Create a new branch for this week's homework (replace `new-branch-name` with the name of your new branch, e.g. 'homework-week1'):
+
+    ```
+    git checkout -b new-branch-name
+    ```
+
+4. Make changes to your local working directory as required by the homework assignment(s).
+
+5. Commit your changes as and when required. Make sure that your commit messages accurately describe what the commit is about.
```
Comment (jasongwartz) [inline fundamentals/homework_pr.md]: I would expand here to say that commit messages like 'my changes' are not okay (I've seen quite a lot of this recently).

We could add an article like [this](https://chris.beams.io/posts/git-commit/) or similar to reinforce it, or just add a note about what a 'good' commit message looks like.

File: fundamentals/homework_pr.md (hunk 1 above)
Comment (jasongwartz) [reply fundamentals/homework_pr.md]: Ah, thank you so much for writing this!
If students can start doing this, it'll be a huge help to homework reviewers.

Comment (remarcmij) [reply fundamentals/homework_pr.md]: Thanks Jason, this was triggered by the message you posted in the class 13 channel.

File: fundamentals/homework_pr.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+5. Commit your changes as and when required. The recommended way to add and commit files is to keep a `.gitignore` file in the root folder of your working directory that lists all the files and directories you want to leave untracked by Git. Now, when you use a `git add .` command, all new, changed and/or deleted files, except those from `.gitignore`, are added to the staging area. 
+
+    ```
+    git add .
+    git commit -m "your commit message"
+    ```
+
+    Make sure that your commit messages are meaningful and accurately describe what the commit is about. Messages such as `my changes` are not okay. If you need to go back in time to review a particular change from the past, the commit messages are the only clues available for finding the relevant commit. For an in-depth discussion on how to write good commit messages, see [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/).
+
```
Comment (remarcmij) [inline fundamentals/homework_pr.md]: I've added something about `.gitignore` that I had planned but forgot. And incorporated the suggestions from Jason.

**Pull request merged**